### PR TITLE
Use Containerfile by default

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
Use the file named "Containerfile" if no build file is specified.  If
not found, fall back to Dockerfile.

This is the initial PR, we probably will have more doc tweaking to do overtime.

```
# cd /tmp
# cp ~/Dockerfile.alpine ./Containerfile
# buildah bud -t tom .
STEP 1: FROM alpine
STEP 2: COMMIT tom
961769676411f082461f9ef46626dd7a2d1e2b2a38e6a44364bcbecf51e66dd4

# mv ./Containerfile ./Dockerfile
# buildah bud -t tom .
STEP 1: FROM alpine
STEP 2: COMMIT tom
961769676411f082461f9ef46626dd7a2d1e2b2a38e6a44364bcbecf51e66dd4

```
Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
I began to play with this on Friday and over the weekend.  LGTM.
But I did fall into a bunch of docs issues and us using the letters "d o c k e r" all over the place.  I think keeping it simple to start and updating the man pages is the way to go.

**Comment by TomSweeneyRedHat** [comment]
I've added a text and touched up the bud page.  Probably other places that we can tweak overtime, but think this works for the first effort.

File: docs/buildah-bud.md [hunk 1]
```diff
@@ -1,28 +1,29 @@
...
 
-If no context directory is specified, then buildah will assume the current working directory as build context, which should contain the Dockerfile.`
+If no context directory is specified, then Buildah will assume the current working directory as build context, which should contain either a Containerfile or a Dockerfile.`
 
 Dockerfiles ending with a ".in" suffix will be preprocessed via CPP(1).  This can be useful to decompose Dockerfiles into several reusable parts that can be used via CPP's **#include** directive.  Notice, a Dockerfile.in file can still be used by other tools when manually preprocessing them via `cpp -E`.
 
 When the URL is an archive, the contents of the URL is downloaded to a temporary location and extracted before execution.
 
-When the URL is an Dockerfile, the Dockerfile is downloaded to a temporary location.
+When the URL is a Dockerfile, the Dockerfile is downloaded to a temporary location.
```
Comment (rhatdan) [inline docs/buildah-bud.md]: Will this work with a Containerfile as well?

**Comment by rhatdan** [comment]
I think you need to make sure you have examples with Dockerfile and Containerfile, in the man page.

File: docs/buildah-bud.md (hunk 1 above)
Comment (TomSweeneyRedHat) [reply docs/buildah-bud.md]: I'm not sure and didn't test it so didn't want to be bold.  I'll see if I can work up a test.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
@vrothberg I've added a new "Dockerfile" test too.  I'd not done that earlier as we do that in 95% of the tests, but probably good to have one specifically.  Holler if I misinterpreted your suggestion.

**Comment by vrothberg** [comment]
> @vrothberg I've added a new "Dockerfile" test too. I'd not done that earlier as we do that in 95% of the tests, but probably good to have one specifically. Holler if I misinterpreted your suggestion.

Thanks, that was it :) I just came to think of another test using a directory as context which containers a Dockerfile and a Containerfile. The build should then select the Containerfile. It won't increase coverage but could prevent potential regressions in the future.

**Comment by TomSweeneyRedHat** [comment]
@vrothberg, good thought, just added another test.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by aric49** [comment]
The status of this PR is closed, did it get merged, or closed out automatically by rh-atomic-bot?

**Comment by TomSweeneyRedHat** [comment]
It's merged.  Buildah now uses files named "Containerfiles" and if not present, then "Dockerfiles" before erroring.  Syntax internally is the same between both files.  Not yet in Podman, but should be the next time Buildah gets vendored there.

**Comment by TomSweeneyRedHat** [comment]
Just to be clear, that's just in upstream ATM, will be in v1.11.3 or v1.12.0, whichever the next release is.

**Pull request closed without merging**